components/authz-service/storage/postgres/rule.go

```diff
@@ -17,6 +18,23 @@ const (
 )
 
 func (p *pg) CreateRule(ctx context.Context, rule *storage.Rule) (*storage.Rule, error) {
+
+	projectsFilter, err := projectsListFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	found := len(projectsFilter) == 0 // empty means all projects allowed
+	if !found {
+		for _, proj := range projectsFilter {
+			if proj == rule.ProjectID {
+				found = true
+			}
+		}
+	}
+	if !found {
+		return nil, fmt.Errorf("project with ID %q not found", rule.ProjectID)
+	}
```

Comment:
Re: previous nit, instead of overwriting `found`, I'd suggest
```suggestion
	if !found && !stringutils.SliceContains(projectsFilter, rule.ProjectID) {
		return nil, fmt.Errorf("project with ID %q not found", rule.ProjectID)
	}
```

Reply:
🤔 maybe there's a better name than `found`, here, too? `allProjectsAllowed := len(...) == 0` maybe?

Reply:
lol I've just created a bunch of comments redundant to @srenatus 😂